WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -336,14 +331,19 @@ extension PostEditor where Self: UIViewController {
             let generator = UINotificationFeedbackGenerator()
             generator.prepare()
 
-            if let error = error {
-                DDLogError("Error publishing post: \(error.localizedDescription)")
-                ActionDispatcher.dispatch(NoticeAction.post(self.uploadFailureNotice(action: action)))
-                generator.notificationOccurred(.error)
-            } else if let uploadedPost = uploadedPost {
+            switch result {
+            case .success(let uploadedPost):
                 self.post = uploadedPost
 
                 generator.notificationOccurred(.success)
+            case .error(let error):
+                guard !self.post.shouldAttemptAutoUpload else {
```

Comment:
Why are we doing this check here?

Reply:
If the post is not marked as `shouldAttemptAutoUpload` I wanna show an error to the user.

But I'm not sure this case will ever happen now that the "Update" button marks the post to be auto uploaded. 🤔 

Maybe makes sense to remove this error triggering logic?

Reply:
So, the problem that I see here is that there are 2 places where we create notices, `PostEditor.uploadPost()` and `PostCoordinator.upload()`:

https://github.com/wordpress-mobile/WordPress-iOS/blob/3848353955553208bae43afa37d2c4e9b68c7283/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L345

https://github.com/wordpress-mobile/WordPress-iOS/blob/3848353955553208bae43afa37d2c4e9b68c7283/WordPress/Classes/Services/PostCoordinator.swift#L266-L268

They're both connected and it's a bit tricky to see when the notice is created.

I was going to suggest that we move the notice-creation to `PostCoordinator.save()` like so:

```swift
prepareToSave(postToSave, automatedRetry: automatedRetry) { result in
	switch result {
	case .success(let post):
		self.upload(post: post) {
			// show notices if necessary
		}
	case .error(let error):
		// show a default notice
		ActionDispatcher.dispatch(NoticeAction.post(Notice(title: "Unknown error")))
		completion?(.error(error))
	}
}
```

But I realized that that method is used in a lot of places. The change in behavior, not to mention the required lifting of `PostEditor.uploadFailureNotice`, might just cause more harm at the moment.

On the other hand, I'm dubious of the usage of `shouldAttemptAutoUpload` here. It requires digging and understanding of how `shouldAttemptAutoUpload` is related to _showing notices_. It's quite complex and could easily break.

## Suggestion

Figuring out why `PostCoordinator` shows notices in the first place is out of the question so we can just go along with it. A minimal impact I can think of is adding a `defaultFailureNotice` parameter:

```swift
func save(_ postToSave: AbstractPost,
		  automatedRetry: Bool = false,
		  defaultFailureNotice: Notice? = nil,
		  completion: ((Result<AbstractPost>) -> ())? = nil) {
	prepareToSave(postToSave, automatedRetry: automatedRetry) { result in
		switch result {
		case .success(let post):
			self.upload(post: post, completion: completion)
		case .error(let error):
			if let notice = defaultFailureNotice {
				ActionDispatcher.dispatch(NoticeAction.post(notice))
			}
			completion?(.error(error))
		}
	}
}
```

With this, we don't need to have the bleed of `shouldAttemptAutoUpload` logic. It's not the best but it's the least amount of effort I can think of at the moment.

Reply:
Done in af8ab6f8a6dfe907084c4ebf2e8bb9c9466fa9fa